package com.winginrian.hibernate.dao;
// Generated 2019-3-19 23:18:13 by Hibernate Tools 5.2.11.Final

import javax.ejb.Stateless;
import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

/**
 * Home object for domain model class Workflowwebserviceoutputmapping.
 * @see com.winginrian.hibernate.dao.Workflowwebserviceoutputmapping
 * @author Hibernate Tools
 */
@Stateless
public class WorkflowwebserviceoutputmappingHome {

	private static final Log log = LogFactory.getLog(WorkflowwebserviceoutputmappingHome.class);

	@PersistenceContext
	private EntityManager entityManager;

	public void persist(Workflowwebserviceoutputmapping transientInstance) {
		log.debug("persisting Workflowwebserviceoutputmapping instance");
		try {
			entityManager.persist(transientInstance);
			log.debug("persist successful");
		} catch (RuntimeException re) {
			log.error("persist failed", re);
			throw re;
		}
	}

	public void remove(Workflowwebserviceoutputmapping persistentInstance) {
		log.debug("removing Workflowwebserviceoutputmapping instance");
		try {
			entityManager.remove(persistentInstance);
			log.debug("remove successful");
		} catch (RuntimeException re) {
			log.error("remove failed", re);
			throw re;
		}
	}

	public Workflowwebserviceoutputmapping merge(Workflowwebserviceoutputmapping detachedInstance) {
		log.debug("merging Workflowwebserviceoutputmapping instance");
		try {
			Workflowwebserviceoutputmapping result = entityManager.merge(detachedInstance);
			log.debug("merge successful");
			return result;
		} catch (RuntimeException re) {
			log.error("merge failed", re);
			throw re;
		}
	}

	public Workflowwebserviceoutputmapping findById(Integer id) {
		log.debug("getting Workflowwebserviceoutputmapping instance with id: " + id);
		try {
			Workflowwebserviceoutputmapping instance = entityManager.find(Workflowwebserviceoutputmapping.class, id);
			log.debug("get successful");
			return instance;
		} catch (RuntimeException re) {
			log.error("get failed", re);
			throw re;
		}
	}
}
